packages/jellyfish-api-core/__tests__/category/loan/takeLoan.test.ts

```diff
@@ -3,279 +3,636 @@ import { GenesisKeys } from '@defichain/testcontainers'
+
+  bobVaultAddr = await bob.generateAddress()
+  bobVaultId = await bob.rpc.loan.createVault({
+    ownerAddress: bobVaultAddr,
+    loanSchemeId: 'scheme'
+  })
+  await bob.generate(1)
+
+  liqVaultAddr = await bob.generateAddress()
+  liqVaultId = await bob.rpc.loan.createVault({
+    ownerAddress: liqVaultAddr,
+    loanSchemeId: 'scheme'
+  })
+  await bob.generate(1)
+
+  mayLiqVaultAddr = await bob.generateAddress()
+  mayLiqVaultId = await bob.rpc.loan.createVault({
+    ownerAddress: mayLiqVaultAddr,
+    loanSchemeId: 'scheme'
+  })
+  await bob.generate(1)
 
-    // collateral token
-    await tGroup.get(0).rpc.loan.setCollateralToken({
-      token: 'DFI',
-      factor: new BigNumber(1),
-      fixedIntervalPriceId: 'DFI/USD'
+  frozenVaultAddr = await bob.generateAddress()
+  frozenVaultId = await bob.rpc.loan.createVault({
+    ownerAddress: frozenVaultAddr,
+    loanSchemeId: 'scheme'
+  })
+  await bob.generate(1)
+
+  emptyVaultAddr = await bob.generateAddress()
+  emptyVaultId = await bob.rpc.loan.createVault({
+    ownerAddress: emptyVaultAddr,
+    loanSchemeId: 'scheme'
+  })
+  await bob.generate(1)
+  await tGroup.waitForSync()
+
+  // deposit on active vault
+  await alice.rpc.loan.depositToVault({
+    vaultId: bobVaultId, from: aliceAddr, amount: '10000@DFI'
+  })
+  await alice.generate(1)
+
+  await alice.rpc.loan.depositToVault({
+    vaultId: bobVaultId, from: aliceAddr, amount: '1@BTC'
+  })
+  await alice.generate(1)
+
+  // deposit on liqVault
+  await alice.rpc.loan.depositToVault({
+    vaultId: liqVaultId, from: aliceAddr, amount: '10000@DFI'
+  })
+  await alice.generate(1)
+
+  // deposit on mayLiqVault
+  await alice.rpc.loan.depositToVault({
+    vaultId: mayLiqVaultId, from: aliceAddr, amount: '10000@DFI'
+  })
+  await alice.generate(1)
+
+  // deposit on frozenVault
+  await alice.rpc.loan.depositToVault({
+    vaultId: frozenVaultId, from: aliceAddr, amount: '10000@DFI'
+  })
+  await alice.generate(1)
+  await tGroup.waitForSync()
+
+  bobVault = await bob.rpc.loan.getVault(bobVaultId) as VaultActive
+  expect(bobVault.loanSchemeId).toStrictEqual('scheme')
+  expect(bobVault.ownerAddress).toStrictEqual(bobVaultAddr)
+  expect(bobVault.state).toStrictEqual('active')
+  expect(bobVault.collateralAmounts).toStrictEqual(['10000.00000000@DFI', '1.00000000@BTC'])
+  expect(bobVault.collateralValue).toStrictEqual(new BigNumber(15000))
+  expect(bobVault.loanAmounts).toStrictEqual([])
+  expect(bobVault.loanValue).toStrictEqual(new BigNumber(0))
+  expect(bobVault.interestAmounts).toStrictEqual([])
+  expect(bobVault.interestValue).toStrictEqual(new BigNumber(0))
+  expect(bobVault.collateralRatio).toStrictEqual(-1) // empty loan
+  expect(bobVault.informativeRatio).toStrictEqual(new BigNumber(-1)) // empty loan
+}
+
+describe('takeLoan success', () => {
+  beforeEach(async () => {
+    await tGroup.start()
+    await alice.container.waitForWalletCoinbaseMaturity()
+    await setup()
```

Comment:
Can replace with a simple single vault setup here. The complete `setup` for each fail cases are heavy and repeated here without being used.